### DISCUSSION TITLE
Server less deployment

### DISCUSSION THREAD

**Discussion by thunder-007** [owncloud/web#9124] [Q&A]
How can I deploy it as server less app in azure app service with files storing in azure blob

**Comment by pascalwengerter** [comment]
> How can I deploy it as server less app in azure app service with files storing in azure blob

@Thunder-007 could you elaborate on your use case a little or provide more context?

The repository you've started the discussion in "only" includes the frontend SPA for the ownCloud web UI, so
- you could host said SPA as a static web app on Azure ([link to the documentation](https://learn.microsoft.com/en-us/azure/static-web-apps/)), but
- you'd need to have a backend (obviously), and I'm not sure whether you've looked into the fact that you can either use [the legacy backend](https://github.com/owncloud/core) or [the new infinite scale project](https://github.com/owncloud/ocis/). With the legacy one I'm pretty sure serverless isn't a viable solution due to the LAMP stack and provisioning a full instance per request (at least that's my understanding of how serverless would work), and depending on the scale you're planning for a small-ish instance of the infinite scale backend may be worth looking into to keep the overall complexity low (since it's not exactly the smale-scale functions you'd usually run serverless, but a microservice-ready solution composed of various parts that need configuration themselves)
- for infinite scale, you could use Azure blob storage with S3 (I suppose?). The documentation lives [here](https://doc.owncloud.com/ocis/next/deployment/general/general-info.html#using-s3-for-blobs)

Looking forward to your answer and hearing more of what you're planning!

**Comment by Lanzernail** [comment]
Deploying OwnCloud in a serverless-like setup on Azure with Azure Blob Storage for file storage is possible, but it does require a few custom steps.

Here's a basic outline of how you can approach it:
1. Deploy OwnCloud to Azure App Service

Package OwnCloud as a PHP app (ZIP or via GitHub).

Create an App Service (Linux) with the appropriate PHP version.

Deploy your OwnCloud files to the App Service (you can use FTP, GitHub, or Azure CLI).

2. Configure Azure Blob Storage as Primary Storage

OwnCloud doesn’t support Azure Blob Storage natively out-of-the-box, but you can use the Flysystem plugin.

OwnCloud uses Flysystem
 as an abstraction layer.

Install and configure the Flysystem Azure Blob adapter.

Update the config.php in OwnCloud to point to your Azure Blob Storage container.

Example config snippet:

'objectstore' => [
    'class' => 'OC\\Files\\ObjectStore\\Flysystem',
    'arguments' => [
        'adapter' => '\\League\\Flysystem\\AzureBlobStorage\\AzureBlobStorageAdapter',
        'client' => [
            'accountName' => 'your-storage-account',
            'accountKey' => 'your-key',
            'container' => 'your-container-name',
        ],
    ],
],


Note: You’ll likely need to add the required libraries via Composer and include them in your App Service deployment.

3. Handle Serverless Aspects

Azure App Service isn’t fully serverless, but you only pay for what you use (especially in Consumption or Premium plans). For true serverless (like Functions), OwnCloud isn’t a great fit due to its monolithic architecture.
